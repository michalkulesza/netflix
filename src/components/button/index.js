import React from "react";
import { Main, Square, Round, Container, Label, Clear, Form } from "./styles/button";

import { ImSpinner8 } from "react-icons/im";

const Button = ({ children, ...restProps }) => {
	return <Main {...restProps}>{children}</Main>;
};

Button.Square = ({ iconScale, inverted, marginRight, children, ...restProps }) => {
	return (
		<Square iconScale={iconScale} inverted={inverted} marginRight={marginRight} {...restProps}>
			{children}
		</Square>
	);
};

Button.Round = ({ label, inverted, dark, marginRight, children, isLoading, ...restProps }) => {
	return (
		<Container marginRight={marginRight}>
			<Round inverted={inverted} dark={dark} isLoading={isLoading} {...restProps}>
				{children}
			</Round>
			{label && <Label>{label}</Label>}
		</Container>
	);
};

Button.Clear = ({ size, padding, children, margin, ...restProps }) => {
	return (
		<Clear size={size} padding={padding} margin={margin} {...restProps}>
			{children}
		</Clear>
	);
};

Button.Form = ({ loading, children, ...restProps }) => {
	return <Form {...restProps}>{loading ? <ImSpinner8 /> : children}</Form>;
};

export default Button;
