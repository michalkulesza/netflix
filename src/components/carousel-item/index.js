import React from "react";
import { Item, Container, Wrapper, Loader, ExpandedSmall } from "./styles/carousel-item";

const CarouselItem = ({ children, ...restProps }) => {
	return <Item {...restProps}>{children}</Item>;
};

CarouselItem.Container = ({ children, ...restProps }) => {
	return <Container {...restProps}>{children}</Container>;
};

CarouselItem.Wrapper = ({ children, ...restProps }) => {
	return <Wrapper {...restProps}>{children}</Wrapper>;
};

CarouselItem.Loader = ({ children, ...restProps }) => {
	return <Loader {...restProps}>{children}</Loader>;
};

CarouselItem.ExpandedSmall = ({ children, ...restProps }) => {
	return <ExpandedSmall {...restProps}>{children}</ExpandedSmall>;
};

export default CarouselItem;
