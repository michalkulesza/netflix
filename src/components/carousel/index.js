import React from "react";
import {
	Main,
	Category,
	Overlay,
	Button,
	Container,
	Wrapper,
	ItemsContainer,
	ItemWrapper,
	Item,
	Loader,
} from "./styles/carousel";

const Carousel = ({ children, ...restProps }) => {
	return <Main {...restProps}>{children}</Main>;
};

Carousel.Category = ({ children, ...restProps }) => {
	return <Category {...restProps}>{children}</Category>;
};

Carousel.Overlay = ({ children, ...restProps }) => {
	return <Overlay {...restProps}>{children}</Overlay>;
};

Carousel.Button = ({ children, ...restProps }) => {
	return <Button {...restProps}>{children}</Button>;
};

Carousel.Container = ({ children, ...restProps }) => {
	return <Container {...restProps}>{children}</Container>;
};

Carousel.Wrapper = ({ children, ...restProps }) => {
	return <Wrapper {...restProps}>{children}</Wrapper>;
};

Carousel.ItemsContainer = ({ children, ...restProps }) => {
	return <ItemsContainer {...restProps}>{children}</ItemsContainer>;
};

Carousel.ItemWrapper = ({ children, ...restProps }) => {
	return <ItemWrapper {...restProps}>{children}</ItemWrapper>;
};

Carousel.Item = ({ children, ...restProps }) => {
	return <Item {...restProps}>{children}</Item>;
};

Carousel.Loader = ({ children, ...restProps }) => {
	return <Loader {...restProps}>{children}</Loader>;
};

export default Carousel;
