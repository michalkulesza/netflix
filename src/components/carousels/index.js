import React from "react";
import { Main } from "./styles/carousels";

const Carousels = ({ children, ...restProps }) => {
	return <Main {...restProps}>{children}</Main>;
};

export default Carousels;
