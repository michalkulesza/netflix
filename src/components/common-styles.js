import { keyframes } from "styled-components/macro";

export const LoadingAnimation = keyframes`
	from {
		background-position-x: 0vw;
	}
	to {
		background-position-x: 100vw;
	}
`;
