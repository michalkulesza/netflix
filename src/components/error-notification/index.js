import React from "react";
import { Main, Wrapper } from "./styles/error-notification";

const ErrorNotification = ({ children, ...restProps }) => {
	return <Main {...restProps}>{children}</Main>;
};

ErrorNotification.Wrapper = ({ children, ...restProps }) => {
	return <Wrapper {...restProps}>{children}</Wrapper>;
};

export default ErrorNotification;
