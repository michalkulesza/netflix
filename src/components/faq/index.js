import React from "react";
import { Main, Container, Title, List, Item, Header, Icon, Body, Text } from "./styles/faq";

const Faq = ({ children, ...restProps }) => {
	return <Main {...restProps}>{children}</Main>;
};

Faq.Container = function ({ children, ...restProps }) {
	return <Container {...restProps}>{children}</Container>;
};

Faq.Title = function ({ children, ...restProps }) {
	return <Title {...restProps}>{children}</Title>;
};

Faq.List = function ({ children, ...restProps }) {
	return <List {...restProps}>{children}</List>;
};

Faq.Item = function ({ children, ...restProps }) {
	return <Item {...restProps}>{children}</Item>;
};

Faq.Header = function ({ children, ...restProps }) {
	return <Header {...restProps}>{children}</Header>;
};

Faq.Icon = function ({ children, ...restProps }) {
	return <Icon {...restProps}>{children}</Icon>;
};

Faq.Body = function ({ children, ...restProps }) {
	return <Body {...restProps}>{children}</Body>;
};

Faq.Text = function ({ children, ...restProps }) {
	return <Text {...restProps}>{children}</Text>;
};

export default Faq;
