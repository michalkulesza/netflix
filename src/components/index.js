export { default as Header } from "./header";
export { default as Navbar } from "./navbar";
export { default as Jumbotron } from "./jumbotron";
export { default as Faq } from "./faq";
export { default as Newsletter } from "./newsletter";
export { default as Footer } from "./footer";
export { default as Button } from "./button";
export { default as Form } from "./form";
export { default as Carousel } from "./carousel";
export { default as Carousels } from "./carousels";
export { default as Item } from "./item";
export { default as ItemExpanded } from "./item-expanded";
export { default as Details } from "./item-details/details";

export { default as ItemDetailsHeader } from "./item-details-header";
export { default as ItemDetailsInfo } from "./item-details-info";
export { ItemDetailsEpisodes } from "./item-details-episodes";
export { ItemDetailsEpisode } from "./item-details-episodes";
