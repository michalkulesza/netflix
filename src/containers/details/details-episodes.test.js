import React from "react";
import { mount } from "enzyme";
import DetailsEpisodesContainer from "./details-episodes";
import { DetailsEpisode, DetailsEpisodes } from "../../components/";
import configureStore from "redux-mock-store";
import * as ReactRedux from "react-redux";
import { BiCaretDown, BiCaretUp } from "react-icons/bi";

let useEffect;
let store = configureStore()({
	fetchEpisodes: {
		data: [],
		isUpdating: false,
	},
});

const mockUseEffect = () => {
	useEffect.mockImplementationOnce(f => f());
};

beforeEach(() => {
	useEffect = jest.spyOn(React, "useEffect");
	mockUseEffect();
	mockUseEffect();
});

describe("<DetailsEpisodesContainer/> with no props", () => {
	const wrapper = mount(
		<ReactRedux.Provider store={store}>
			<DetailsEpisodesContainer />
		</ReactRedux.Provider>
	);

	it("should match the snapshot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});
});

describe("<DetailsEpisodesContainer/> with props and updating", () => {
	store = configureStore()({
		fetchEpisodes: {
			data: [],
			isUpdating: true,
		},
	});

	const mockItem = {};

	const wrapper = mount(
		<ReactRedux.Provider store={store}>
			<DetailsEpisodesContainer item={mockItem} />
		</ReactRedux.Provider>
	);

	it("should match the snapshot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("should contain <DetailsEpisode.WrapperLoading />", () => {
		expect(wrapper.contains(DetailsEpisode.WrapperLoading)).toBeTruthy();
	});
});

describe("<DetailsEpisodesContainer/> with props", () => {
	store = configureStore()({
		fetchEpisodes: {
			data: [
				{
					id: 1,
					episode_number: 9,
					still_path_300: "/",
					name: "Test",
					overview: "Lorem ipsum dolor, sit amet consectetur adipisicing elit. Vel, illo.",
				},
				{
					id: 2,
					episode_number: 9,
					still_path_300: "/",
					name: "Test2",
					overview: "Vel, illo.",
				},
			],
			isUpdating: false,
		},
	});

	const mockItem = {
		details: {
			number_of_seasons: 11,
		},
	};

	const wrapper = mount(
		<ReactRedux.Provider store={store}>
			<DetailsEpisodesContainer item={mockItem} />
		</ReactRedux.Provider>
	);

	it("should match the snapshot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("should not contain <DetailsEpisode.WrapperLoading />", () => {
		expect(wrapper.contains(<DetailsEpisode.WrapperLoading />)).toBeFalsy();
	});

	it("should render list of seasons", () => {
		expect(wrapper.find(DetailsEpisodes.Season).length).toBe(mockItem.details.number_of_seasons);
	});

	it("should render list of episodes", () => {
		expect(wrapper.find(DetailsEpisode.Wrapper).length).toBe(store.getState().fetchEpisodes.data.length);
	});

	it("should display correct icon depending on clicked button", () => {
		expect(wrapper.contains(<BiCaretDown />)).toBeTruthy();
		wrapper.find(DetailsEpisodes.SeasonsButton).simulate("mouseDown");
		expect(wrapper.contains(<BiCaretUp />)).toBeTruthy();
	});
});
