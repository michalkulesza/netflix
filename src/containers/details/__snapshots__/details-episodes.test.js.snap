// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailsEpisodesContainer/> with no props should match the snapshot 1`] = `""`;

exports[`<DetailsEpisodesContainer/> with props and updating should match the snapshot 1`] = `"<div class=\\"details-episodes__Episodes-q2f4d2-0 kNwKbG\\"><div class=\\"details-episodes__EpisodesHeader-q2f4d2-1 hrxrGP\\">Episodes<div class=\\"details-episodes__EpisodesSeasons-q2f4d2-2 gwFCAB\\"><button data-testid=\\"SeasonsButton\\" class=\\"details-episodes__EpisodesSeasonsButton-q2f4d2-3 dcCfkk\\"><span>Season 1</span></button><ul class=\\"details-episodes__EpisodesSeasonsList-q2f4d2-4 eJeDmD\\"><li class=\\"details-episodes__EpisodesSeason-q2f4d2-5 fQVBUV\\">Season 1</li></ul></div></div> <div class=\\"details-episodes__EpisodesList-q2f4d2-6\\"><div class=\\"details-episodes__EpisodeWrapperLoading-q2f4d2-8 MufwE\\"></div></div></div>"`;

exports[`<DetailsEpisodesContainer/> with props should match the snapshot 1`] = `"<div class=\\"details-episodes__Episodes-q2f4d2-0 kNwKbG\\"><div class=\\"details-episodes__EpisodesHeader-q2f4d2-1 hrxrGP\\">Episodes<div class=\\"details-episodes__EpisodesSeasons-q2f4d2-2 gwFCAB\\"><button data-testid=\\"SeasonsButton\\" class=\\"details-episodes__EpisodesSeasonsButton-q2f4d2-3 dcCfkk\\"><span>Season 1</span><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M11.998 17L18.998 9 4.998 9z\\"></path></svg></button><ul class=\\"details-episodes__EpisodesSeasonsList-q2f4d2-4 eJeDmD\\"><li class=\\"details-episodes__EpisodesSeason-q2f4d2-5 fQVBUV\\">Season 1</li><li class=\\"details-episodes__EpisodesSeason-q2f4d2-5 fQVBUV\\">Season 2</li><li class=\\"details-episodes__EpisodesSeason-q2f4d2-5 fQVBUV\\">Season 3</li><li class=\\"details-episodes__EpisodesSeason-q2f4d2-5 fQVBUV\\">Season 4</li><li class=\\"details-episodes__EpisodesSeason-q2f4d2-5 fQVBUV\\">Season 5</li><li class=\\"details-episodes__EpisodesSeason-q2f4d2-5 fQVBUV\\">Season 6</li><li class=\\"details-episodes__EpisodesSeason-q2f4d2-5 fQVBUV\\">Season 7</li><li class=\\"details-episodes__EpisodesSeason-q2f4d2-5 fQVBUV\\">Season 8</li><li class=\\"details-episodes__EpisodesSeason-q2f4d2-5 fQVBUV\\">Season 9</li><li class=\\"details-episodes__EpisodesSeason-q2f4d2-5 fQVBUV\\">Season 10</li><li class=\\"details-episodes__EpisodesSeason-q2f4d2-5 fQVBUV\\">Season 11</li></ul></div></div> <div class=\\"details-episodes__EpisodesList-q2f4d2-6\\"><div class=\\"details-episodes__EpisodeWrapper-q2f4d2-7 bXGNmu\\"><div class=\\"details-episodes__Episode-q2f4d2-9 kMXxxi\\"><div class=\\"details-episodes__EpisodeNum-q2f4d2-10 fBBkLn\\">9</div><img src=\\"/\\" alt=\\"Episode preview\\" class=\\"details-episodes__EpisodeImage-q2f4d2-11 keHPpr\\"><div class=\\"details-episodes__EpisodeMain-q2f4d2-12 bnQqwN\\"><div class=\\"details-episodes__EpisodeHalf-q2f4d2-13 dRGDpb\\"><div class=\\"details-episodes__EpisodeTitle-q2f4d2-14 jXcepZ\\">Test</div><div class=\\"details-episodes__EpisodeTime-q2f4d2-15\\">57m</div></div><div class=\\"details-episodes__EpisodeHalf-q2f4d2-13 dRGDpb\\"><div class=\\"details-episodes__EpisodeDescription-q2f4d2-16 fntMBP\\">Lorem ipsum dolor, sit amet consectetur adipisicing elit. Vel, illo.</div></div></div></div></div><div class=\\"details-episodes__EpisodeWrapper-q2f4d2-7 bXGNmu\\"><div class=\\"details-episodes__Episode-q2f4d2-9 kMXxxi\\"><div class=\\"details-episodes__EpisodeNum-q2f4d2-10 fBBkLn\\">9</div><img src=\\"/\\" alt=\\"Episode preview\\" class=\\"details-episodes__EpisodeImage-q2f4d2-11 keHPpr\\"><div class=\\"details-episodes__EpisodeMain-q2f4d2-12 bnQqwN\\"><div class=\\"details-episodes__EpisodeHalf-q2f4d2-13 dRGDpb\\"><div class=\\"details-episodes__EpisodeTitle-q2f4d2-14 jXcepZ\\">Test2</div><div class=\\"details-episodes__EpisodeTime-q2f4d2-15\\">57m</div></div><div class=\\"details-episodes__EpisodeHalf-q2f4d2-13 dRGDpb\\"><div class=\\"details-episodes__EpisodeDescription-q2f4d2-16 fntMBP\\">Vel, illo.</div></div></div></div></div></div></div>"`;
